### ISSUE TITLE
Wandering traders spawn way too often with the improved wandering traders option disabled

### ISSUE THREAD

**Issue by ndrp** [svenhjol/Charm#473]
Versions
* Charm version - 2.2.0
* Minecraft version - 1.6.4

Wandering traders spawn way too often with the improved wandering traders option disabled. According to the Minecraft Wiki it should take 10 Minecraft days on average for a wandering trader to appear. I haven't even been playing the world for 10 days and already had 4 traders appear.

Edit: The "wandering traders spawn more frequently after one Minecraft day" option is also set to false, although it shouldn't matter since wandering trader improvements is overall set to false, I think?

**Comment by F43nd1r** [comment]
> On **average**, it takes 10.411 Minecraft days for a wandering trader to spawn.
> -Minecraft wiki (emphasis mine)

It is possible for a trader to spawn every day, it's just unlikely. 

Reopen if your sample size is statistically significant or if trader spawns follow a rigid pattern (one spawns e.g. every day or every second day 100% of the time).

**Comment by ndrp** [comment]
> It is possible for a trader to spawn every day, it's just unlikely.

Thanks for the lesson on statistics, I am aware of that though. It's my fault for understating, this isn't the first world where I have noticed a lot of wandering traders spawning since installing Charm.

Now, I don't know much about modding Minecraft or Mixins yet, so forgive me if I'm wrong, but I took a look at the code in [WanderingTraderManagerMixin.java](https://github.com/svenhjol/Charm/blob/master/src/main/java/svenhjol/charm/mixin/WanderingTraderManagerMixin.java) and if I may hazard a guess, could it be that in

``` 
private int hookRandomCheck(Random random, int i) {
        return WanderingTraderImprovements.shouldSpawnFrequently() ? 0 : random.nextInt(1); // 10 is vanilla random value
    }
```
`random.nextInt(1)` is supposed to be `random.nextInt(10)` ?

**Comment by F43nd1r** [comment]
That is true. I've checked the source before closing, but I only checked the forge code, sorry.

Assigning @svenhjol

**Comment by F43nd1r** [comment]
Fixed in [v2.3.0](https://github.com/svenhjol/Charm/releases/tag/2.3.0)
